buildSrc/src/main/kotlin/buildsrc/convention/mockk-publishing.gradle.kts

```diff
@@ -47,21 +41,35 @@ val mavenName: String by project.extra
 val mavenDescription: String by project.extra
 val localrepo: String by project
 
-publishing {
+extensions.getByType<PublishingExtension>().apply {
+//publishing {
     repositories {
         // publish to local dir, for testing
         maven(rootProject.layout.projectDirectory.dir(localrepo)) {
             name = "LocalRepo"
         }
 
-        /*maven {
-            url = uri(sonatypeRepositoryReleaseUrl)
-            credentials {
-                username = ossrhUsername.get()
-                password = ossrhPassword.get()
+        if (ossrhUsername.isPresent && ossrhPassword.isPresent) {
+            maven("https://oss.sonatype.org/content/repositories/snapshots/") {
+                name = "SonartypeStaging"
+                credentials {
+                    username = ossrhUsername.get()
+                    password = ossrhPassword.get()
+                }
+            }
+
+            if (!isSnapshotVersion.get()) {
+                maven("https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
```

Comment:
Please check this guide https://central.sonatype.org/publish/publish-gradle/.
I have doubts that this is right URL.
I will try to deal with required secrets and set them in GH settings:
 - secrets.GRADLE_SIGNING_KEY_ID
 - secrets.GRADLE_SIGNING_PASSWORD
 - secrets.GRADLE_SIGNING_SECRET_KEY_RING_FILE
 - secrets.OSS_SONATYPE_USERNAME
 - secrets.OSS_SONATYPE_PASSWORD

Reply:
Thanks @oleksiyp!

Would you consider [granting @Raibaz ownership](https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-peoples-access-to-your-organization-with-roles/maintaining-ownership-continuity-for-your-organization)? Then he could also manage the secrets.

> I have doubts that this is right URL.


MockK was published before 2021 so still uses the 'legacy host', but the docs mostly use the new URLs.

> Note: As of February 2021, all new projects began being provisioned on https://s01.oss.sonatype.org/. If your project is not provisioned on https://s01.oss.sonatype.org/, you will want to reference the legacy host https://oss.sonatype.org/.

The current MockK config uses the same URL and it works okay

https://github.com/mockk/mockk/blob/1419517e0de42c6148d0fb325467d16797ed3e53/buildSrc/src/main/kotlin/buildsrc/convention/mockk-publishing.gradle.kts#L15-L21

Reply:
Not sure you will need other secrets... and not sure you need them for something else... security is always compromise of comfort. So no 😄  but remember that I am still alive and happy to help 😄

Reply:
Afterall if @Raibaz feels that more access is required happy to understand the reasoning and cooperate.

There is a bigger issue where to spend $4k on https://opencollective.com/mockk. Previously I thought it was too small an amount to spend paying contributors... but now it is not that obvious

Reply:
I'm perfectly fine with the access I have now, I don't think I need more :)

After all, this should be a one-shot operation and we should be ok with secrets now, I don't think I'll need access to change them anytime soon.